haystack/document_stores/in_memory/document_store.py

```diff
@@ -132,7 +318,36 @@ def write_documents(self, documents: List[Document], policy: DuplicatePolicy = D
                     logger.warning("ID '{document_id}' already exists", document_id=document.id)
                     written_documents -= 1
                     continue
+
+            # Since the statistics are updated in an incremental manner,
+            # we need to explicitly remove the existing document to revert
+            # the statistics before updating them with the new document.
+            if document.id in self.storage.keys():
```

Comment:
we need to add a safeguard here for the case `policy == DuplicatePolicy.SKIP`

Reply:
I think this logic is already captured by L314? But I agree that we can make this more explicit

Reply:
you're right, I overlook that

Reply:
Ok! Then I will keep it as is